Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -282,6 +283,10 @@ public async Task<Page> PopAsyncInner(bool animated, bool fast)
 
 		async Task<Page> INavigationPageController.RemoveAsyncInner(Page page, bool animated, bool fast)
 		{
+			if (_ignorePopCall) 
```

Comment:
I agree this doesn't quite feel like the place to fix.  *RemoveAsyncInner* being called twice on the same page seems like the bug.  I wonder if we can fix this purely inside the iOS *NavigationRenderer* .  

For example returning from 
*NavigationRenderer.UpdateFormsInnerNavigation* if _ignorePopCall true

and then setting _ignorePopCall to true inside

*protected virtual async Task<bool> OnPopViewAsync(Page page, bool animated)*

 I also pushed a UI Test that we can use to test this as well